arbos/programs/programs.go

```diff
@@ -283,6 +288,70 @@ func (p Programs) getProgram(codeHash common.Hash, time uint64) (Program, error)
 	return program, err
 }
 
+// SaveActiveProgramToWasmStore is used to save active stylus programs to wasm store during rebuilding
+func (p Programs) SaveActiveProgramToWasmStore(statedb *state.StateDB, codeHash common.Hash, code []byte, time uint64, debugMode bool, rebuildingStartBlockTime uint64) error {
+	params, err := p.Params()
+	if err != nil {
+		return err
+	}
+
+	program, err := p.getActiveProgram(codeHash, time, params)
```

Comment:
I think you can use "rebuildingStartBlockTime" here as well, which means you don't need to get latestHeader.Time at all in the calling function. You only need the latest block in the calling function if stateDb for RebuildingStartBlock does not exist

Reply:
I agree, `rebuildingStartBlockTime` and `time` are equal when first starting out and in the context of this PR where rebuilding is not done parallelly there's no need of latestHeader.Time. 

But this implementation is to future-proof when we'll have a parallel version for rebuilding, and for non-archive nodes we might not have stateDb for RebuildingStartBlock.